File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        height = 0
        res = []
        def dfs(root,res,currHeight):
            if not root:
                return res
            nonlocal height
            if currHeight > height:
                res.append(root.val)
                height = currHeight
            dfs(root.right,res,currHeight+1)
            dfs(root.left,res,currHeight+1)
            return res

        return dfs(root,res,1)

        

File: Data Structures & Algorithms/binary-tree-postorder-traversal/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[]}
     */
    postorderTraversal(root) {
        const res = [];
        function getResult(node){
            if(!node) return;
            getResult(node.left);
            getResult(node.right);
            res.push(node.val);
        }
        getResult(root);
        return res;
    }
}
